app/components/lead_providers/partnerships/challenged_banner.html.erb

```diff
@@ -0,0 +1,4 @@
+<%= govuk_notification_banner title: "Important", classes: %w(govuk-notification-banner--red) do %>
+  <p class="govuk-!-margin-top-0"><strong><%= t '.header' %></strong></p>
+  <p><%= t ".content", reason: challenge_reason %>
```

Comment:
It is not always about re-using, but rather ability to reuse. This template is now decoupled from its content. I can easily test that the text appears in the oputput without actually knowing nor hard-coding the exact text.

Reply:
Well, you can, but is it worth testing that this component rendered those strings? This component is remarkably simple - a header paragraph and a content paragraph (and a title I guess). No conditionals, no loops, no edge cases.

Are we testing that `govuk_notification_banner` is gonna display the content it gets?

I guess the one thing that might be worth testing is that it displays the correct partnership text, the rest is hard-coded.

And decoupling means that it's a lot harder to verify that the text you get makes sense from reading code, because you kind of need the exact text for that, and you need to either put it together in your head or run the code to see it. Not to mention, it will make it harder for designers to find the content to edit.

Reply:
> is it worth testing that this component rendered those strings

I'd say it is. They might be hard-coded now but if that was at any point changed, by mistake or on purpose our tests should pick this up. If we display the banner without the header, this would be flagged as a bug and that test is potentially saving us from this (with a very low time cost, unlike feature test).

> it's a lot harder to verify that the text you get makes sense from reading code

Code and text are different things - you should not need to read the code to verify the text and vice versa. The translation labels should be clear enough for templates to make sense. I am usually completely lost reading the templates because there is just too much text there.

Coupling is the main villain of software maintainability and should be battled almost at any cost in a long-term projects as it significantly increases accidental complexity. Decoupling is not easy (especially in rails - which actively enforces coupling). I agree it will be a bit more tricky for designers to find the content to edit - but it also not a rocket science and they will be able to do this after 1 or 2 short pair programming sessions.

Generally speaking, I think it is just a matter of personal preference and getting used to it. Working on systems written in both ways I always found the ones with enforced i18n translation in the views not only much simpler to follow, but also much better tested, usually with full CI/CD pipeline from PR merge straight into production code.

Reply:
Hehe, in that case let's i18n the banner title too :)